src/engine/Drawing/SpriteSheet.ts

```diff
@@ -133,6 +133,35 @@ export class SpriteSheetImpl {
         throw new Error('Invalid index: ' + index);
       }
    }
+
+   /**
+    * Get an animation with bespoke sprite coordinates. This is useful if the SpriteSheet is
+    * packed and not a uniform width or height. The resulting [[Animation]] will have the height and width of the
+    * largest dimension (width, height) from among the sprite coordinates
+    * @param engine 
+    * @param spriteCoordinates 
+    * @param speed 
+    */
+   public getSpriteByCoords(engine: Engine, spriteCoordinates: ISpriteArgs[], speed: number): Animation {
```

Comment:
I would expect a method with this name to return a `Sprite`, not an `Animation`.  

I would probably change this method to just return a `Sprite` array, then add a static method (or perhaps constructor) to `Animation` that creates an `Animation` from an arbitrary `Sprite` array.

Reply:
on second look, it looks like that constructor already exists -- so just changes this to return a `Sprite` array.

Reply:
Scratch all that, I just saw this `SpriteSheet`, not `Sprite`. just rename this to `getAnimationByCoords`.

Reply:
@alanag13 Updated per feedback